api/rootcerts.go

```diff
@@ -0,0 +1,126 @@
+package api
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+)
+
+// CertConfig determines where LoadCACerts will load certificates from.
+type CertConfig struct {
+	// CAFile is a path to a PEM-encoded certificate file or bundle. Takes
```

Comment:
You know, I wonder if we can do something better (from a UX perspective) and just greedily take all three of them. :-) 

The pool itself does deduplication: https://cs.opensource.google/go/go/+/refs/tags/go1.23.2:src/crypto/x509/cert_pool.go;l=199-202

So we could do something like:

```go
pool, err := x509.SystemCertPool()
...
if CAFile != "" {
  pool.AddCerts(...)
}

&c
```

and get more CAs, rather than fewer. I think this would _technically_ break anyone who expected only a single CA to be provisioned, but I don't know how often that really occurs versus just wanting all certs to be taken. My 2c!

Reply:
I think this is a good idea in the long run. I guess that would mean there is no more priority among sources since all of them are loaded if provided. For users who expect a single CA or depend on the priority stuff, do you think it would make sense to add an optional parameter for prioritizing a specific source? If it's left empty, all sources would be loaded, but if set, only that particular certificate source would be used.

Reply:
@cipherboy I found a edge case not being accounted for in the previous rootcerts. If in `CAPath`, there was a non-certificate PEM, the tests would fail with an error: `client_test.go:400: error reading environment: Error loading CA Path: Couldn't parse PEM in: /Users/fatima/openbao/api/test-fixtures/keys/key.pem` I fixed this in commit 66a85bbb0c5ed0e49add3f0b9da746fa799087fa by adding a check for contents of the PEM file. 

The reason the test wasn't failing previously was because CAPath has the lowest priority and wasn't tested independently before:
https://github.com/openbao/openbao/blob/e6ed492e570d7783c02ae7556e3f8fdba9f96ac5/api/client.go#L297-L301

Reply:
Nice catch! :D 

I agree, that's a good thing to avoid as some systems often have non-cert files in their CA directory. E.g., `/etc/pki/ca-trust/extracted/pem/` on my Fedora box has a `README` in it, which is neither cert or key. 

---

Here is good because its local to where the code and context is :-)